projects/102_homework/src/main.c

```diff
@@ -0,0 +1,45 @@
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#define DELAY_COUNTER_A_MS 500
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void soft_timer_callback(SoftTimerId timer_id, void *context) {
+  Counters *increment = context;
+
+  increment->counter_a++;
+
+  LOG_DEBUG("Counter A: %i\n", increment->counter_a);
+
+  if (increment->counter_a % 2 == 0) {
+    increment->counter_b++;
+
+    LOG_DEBUG("Counter B: %i\n", increment->counter_b);
+  }
+
```

Comment:
Nit: there seems to be a lot of whitespace, which isn't necessarily a problem but makes things harder to read. I would put a bit more thought into how you group things together (Fix if you'd like)